lib/tasks/backend_api.rake

```diff
@@ -0,0 +1,6 @@
+namespace :backend_api do
+  desc 'clear backend_apis that do not belongs to any service'
+  task :destroy_orphans => :environment do
+    BackendApi.where('id NOT IN (SELECT DISTINCT(backend_api_id) FROM backend_api_configs)').destroy_all
```

Comment:
As @macejmic said in the tchat, you could use `DestroyAllDeletedObjectsWorker` (needs a slight change to make it work)